### DISCUSSION TITLE
优化4.2 分流指数算法优化

### DISCUSSION THREAD

**Discussion by welllom** [famland/website#126] [Ideas]
# 优化4.2  分流指数算法优化

当前农场流速规则：

**农场分流指数=农场增量共享值/YLE所有农场增量共享值之和**

**分流速=总流速×农场分流指数**

目前以上规则主要问题体现在农场具有先发优势，对于后期进场玩家所建农场发展不利，如果YLE生态不能拥有足够的激励动力，没有让新农场发展有利可图。那么，这个生态就会是死水一潭。如何既保障建新农场有足够的诱惑力，又不至于破坏生态和老农场利益。

要解决这个问题，有一个原则性的大前提，那就是新农场的付出必然要大于老农场才有可能。毕竟，老农场花费的时间，已经比新农场长了。而时间，在元宇宙新经济中是价值不菲的资源。那就意味着，新农场要有更大的资本投入，以及更好的运气使然。

**那么新的**

**农场分流速=总流速 × （农场分流指数/YLE所有农场分流指数之和）**

**农场分流指数=  农场平均共享值 × 流速系数**

农场**平均共享值：**

共享农场Famland有效数量＜30时，

共享农场平均共享值=当前共享农场总共享值/30

共享农场Famland有效数量≥30时

共享农场平均共享值=当前共享农场总共享值/农场Famland有效数量

**Famland有效数量是指：activeted=1且基本田做市本金大于等于100U的数量（之前农场Famland的数量只考虑了activeted=1，需要补上做市本金大于100U的条件重新核算，也就是即便Famland激活做市基本田了，但是如果撤资后也是不计算在内的）**

**流速系数= 基本系数 + 规模系数 + 运气系数  （流速系数最高值为3，基本系数、规模系数、运气系数峰值均为1），**

流速指数、以及基本系数、规模系数、运气系数**这三个系数均为农场属性参数。**

**基本系数：**

以30块有效famand的农场标准，基本系数为1，最高值为1，农场Famland有效数量每少一块，基本系数减少0.1，也就是当农场Famland有效数量减少到20个后，基本系数将降为0。

**规模系数：**

Famland的有效数量增长分为5个阶段：

当YLE所有famland有效数量在1万以下时，农场每个Famland规模系数计算0.005   rffrefrFamland总量在1万至10万以下时，农场每个Famland规模系数计算0.0025

Famland总量在10万-50万时，农场每个Famland规模系数计算0.00125

Famland总量在50万-300万时，农场每个Famland规模系数计算0.0005

Famland总量在300万以上时，农场每个Famland规模系数计算0.000167

不论农场在任何发展阶段，农场的规模系数最大值峰值为1后不再增加。

每个耕种周期，系统要计算一次当前YLE内，每个activeted=1且基本田做市本金大于等于100U的Famland数量，并按照以上规则重新计算一次各个农场的规模系数。

这样修正后，更能发挥流速系数的阶段性功效，激励新老农场发展新田的动力。也就是说，新农场如果最高投入10万U，200块基本田，就能打造一个流速指数为2的共享农场。一个60块基本田的农场，流速系数就只有1.3。这个增加新田的激励效应就相当明显了。但是，丝毫不会破坏整个Famland的生态。

**运气系数：**

Famland的运气系数主要是通过为Famland栽种CryptoTree随机获得的一种算法加成，这种运气有一定博弈和游戏性。在每个耕种周期，只有加入共享农场的Famalnd栽种八宝树后都会有几率为所在农场提升运气系数。而目前已经在农场为Famland栽种过树的农场是没有这个运气系数的。

栽种树提升运气系数的规则：9**0%的几率栽种八宝树为农场提升的运气系数为0，10%的几率，栽种树将会为所在农场提升0.001-0.009的运气系数。运气系数最高值为1。（随机的算法和参照无常滑点的算法机制，根据hash计算得出提升数值）**

运气系数，主要是为后期大资本、大团队介入而设计。早期的农场，无论怎么发展，资本总是有限的。扩大农田数量，在前期的一定阶段中，还有可能。但越往后，越困难。而要栽种树，所花费的代价就更大了。要达到最高值1，除了看运气好坏，还至少需要500-1000树。即便按照现阶段1000YL1株，当YL价回归到1U时，也需要50-100万U。而到时树价格一定会水涨船高，这样的投资规模，早期任何农场都不可能具备。

**Comment by yancanlou** [comment]
相关系数与实际田数非正比例相关，还受限。不如逢30块保证了基本系数就可以，再建一个农场比加什么系数划算多了。
